functions/linear_section.h

```diff
@@ -0,0 +1,51 @@
+#ifndef FUNCTION_LINEAR_SECTION_H
+#define FUNCTION_LINEAR_SECTION_H
+
+#include <algorithm>
+
+// Usage: LinearSectionF<POSITION, FRACTION>
+// POSITION: FUNCTION position on the blade, 0-32768
+// FRACTION: FUNCTION how much of the blade to light up, 0 = none, 32768 = all of it
+// return value: FUNCTION
+// creates a "block" of pixels at POSITION taking up FRACTION of blade
+
+class BladeBase;
+template<class POSITION, class FRACTION>
+class LinearSectionF {
+public:
+  FunctionRunResult run(BladeBase* base) {
+    pos_.run(base);
+    FunctionRunResult ret = RunFunction(&fraction_, base);
+    
+    num_leds_ = base->num_leds();
+    int fraction = fraction_.getInteger(0);
+    if (fraction == 32768) {
```

Comment:
This optimization doesn't work for LinearSectionF and should be removed.
(Just because fraction is 1.0, doesn't mean that it covers the whole blade...)

Reply:
removed